Tag ESI resources for easier cleanup

DanNiESh opened the pull request:
This PR fixes this [issue](https://github.com/innabox/issues/issues/209). I used the following ansible playbook to test the new changes:
```
---
- hosts: localhost
  gather_facts: false
  vars:
    test_network_suffix: "test-tagging"
    test_tags: ['o-sac', 'cluster-test']

  tasks:
    - name: Test network creation with tagging
      ansible.builtin.include_role:
        name: massopencloud.esi.network
      vars:
        network_suffix: "{{ test_network_suffix }}"
        network_state: present
        network_tags: "{{ test_tags }}"

    - name: Verify network was created with tag
      ansible.builtin.command: >-
        openstack network show "network-{{ test_network_suffix }}" -f json
      register: network_info
      changed_when: false

    - name: Display network tags
      ansible.builtin.debug:
        msg: "Network tags: {{ (network_info.stdout | from_json).tags }}"

    - name: Create floating ip
      ansible.builtin.include_role:
        name: massopencloud.esi.floating_ip
      vars:
        floating_ip_name: "{{ test_network_suffix }}-floating-ip"
        floating_ip_state: present
        floating_ip_tags: "{{ test_tags }}"

    - name: Verify floating ip was created with tag
      ansible.builtin.command: >-
        openstack floating ip show "{{ allocate_floating_ip_result }}" -f json
      register: floating_ip_info
      changed_when: false

    - name: Display floating ip tags
      ansible.builtin.debug:
        msg: "Floating ip tags: {{ (floating_ip_info.stdout | from_json).tags }}"

    - name: Delete floating ip
      ansible.builtin.command: >-
        openstack floating ip delete "{{ allocate_floating_ip_result }}"
      register: floating_ip_delete_result
      changed_when: false

    - name: Delete test network
      ansible.builtin.include_role:
        name: massopencloud.esi.network
      vars:
        network_suffix: "{{ test_network_suffix }}"
        network_state: absent
```

tzumainn commented:
> > The individual playbooks for floating_ip/l2/l3 all use `network_tag`; any reason not to make it `subnet_tag`, `router_tag`, etc?
> 
> I would prefer that we use the same tag name in all cases (because we are representing the same information).

Okay, I should have fully thought through my reasoning. I think that the tags to each of these playbooks should default to an empty list; and I have the impression that the parameters listed in defaults need to be prefixed with the role in order to pass the lint checks.

DanNiESh commented:
- Updated the tag to a list. 
- The tags for network/floating ip are set to `['o-sac', <cluster_name>]` in cloudkit playbooks. 
- Renamed the `network_tag` to `floating_ip_tag`, `l2_network_tag` and `l3_network_tag` to be prefixed with the roles. 
- Tag in esi playbooks are default to `[]`
